### PR TITLE
Fix #11464: Exported PDFs don't have correct size and added open destination folder on export option

### LINKED ISSUE #11464: [MU4 Issue] Exported parts don't have correct PDF size
**Describe the bug**
Exported parts don't have correct PDF size.

**To Reproduce**
1. Format > Page Settings, make a custom size page. Eg 7.75" x 5.5", landscape.
2. Export a part to a PDF
3. The PDF is not 7.75"x5.5", but 8.26"x11.69. The score is in the upper left.

**Expected behavior**
The PDF size should match the page size, as it did in MS3.

**Platform information**
 - OS: linux

version: OS: Ubuntu 20.04.4 LTS, Arch.: x86_64, MuseScore version (64-bit): 4.0.0-2059995262, revision: github-musescore-musescore-cafefd7

### PR DESCRIPTION
**Pull request by HemantAntony**
Resolves: #11464 

- [x] I signed [CLA](https://musescore.org/en/cla)
- [x] I made sure the code in the PR follows [the coding rules](https://github.com/musescore/MuseScore/wiki/CodeGuidelines)
- [x] I made sure the code compiles on my machine
- [x] I made sure there are no unnecessary changes in the code
- [x] I made sure the title of the PR reflects the core meaning of the issue you are solving
- [x] I made sure the commit message(s) contain a description and answer the question "Why do those changes fix that particular issue?" or "Why are those changes really necessary as improvements?"
- [x] I made sure the commit message title starts with "fix #424242:" if there is a related issue
- [ ] I created the test (mtest, vtest, script test) to verify the changes I made

### CODE REVIEW

**Comment by HemantAntony** [comment]
@RomanPudashkin This PR contains a fix for a critical bug #11464
@Tantacrul I hope the checkbox for `Open destination folder on export` is good

**Comment by Tantacrul** [comment]
I'll test when the checks are completed

File: src/importexport/imagesexport/internal/pdfwriter.cpp [hunk 1]
```diff
@@ -115,10 +115,11 @@ mu::Ret PdfWriter::writeList(const INotationPtrList& notations, io::Device& dest
     return true;
 }
 
-void PdfWriter::preparePdfWriter(QPdfWriter& pdfWriter, const QString& title) const
+void PdfWriter::preparePdfWriter(QPdfWriter& pdfWriter, const QString& title, const QSizeF size) const
```
Comment (RomanPudashkin) [inline src/importexport/imagesexport/internal/pdfwriter.cpp:118]: const QSizeF **&** size

**Review by RomanPudashkin — APPROVED** [review]

File: src/importexport/imagesexport/internal/pdfwriter.h [hunk 2]
```diff
@@ -41,7 +41,7 @@ class PdfWriter : public AbstractImageWriter
     Ret writeList(const notation::INotationPtrList& notations, io::Device& destinationDevice, const Options& options = Options()) override;
 
 private:
-    void preparePdfWriter(QPdfWriter& pdfWriter, const QString& title) const;
+    void preparePdfWriter(QPdfWriter& pdfWriter, const QString& title, const QSizeF size) const;
```
Comment (Jojo-Schmitz) [inline src/importexport/imagesexport/internal/pdfwriter.h:44]: here too I guess

**Comment by HemantAntony** [comment]
@RomanPudashkin Should I be concerned about the failing Utests. Its a first for me

**Comment by RomanPudashkin** [comment]
> @RomanPudashkin Should I be concerned about the failing Utests. Its a first for me

The fix will be quite simple

1. you should enable this build option
Screenshot 2022-05-01 at 12 26 48

2. after the previous step, build your project and you will notice a compilation error

3. fix the error and push changes

**Comment by Jojo-Schmitz** [comment]
With step 3 being the hard part ;-)

**Comment by RomanPudashkin** [comment]
> With step 3 being the hard part ;-) as the error message is pretty cryptic. To me at least...

This problem is quite easy to solve. We have ProjectConfigurationMock that inherits IProjectConfiguration. Hemant changed the signature of the IProjectConfiguration::exportScores method (added one more argument). So, we need to changed the signature of ProjectConfigurationMock::exportScores in the same way

**Comment by Tantacrul** [comment]
OK, tested. This is working now. Thanks!

(On a separate note, I remember that there is another issue for allowing users to export parts without needing to open them first, correct?)

**Comment by HemantAntony** [comment]
I think #11356

**Comment by HemantAntony** [comment]
I have a potential fix for #11356 . I'll add that too

**Comment by RomanPudashkin** [comment]
> @Tantacrul @RomanPudashkin I have a potential fix for #11356 . I'll add that too

In a separate PR, please

**Comment by RomanPudashkin** [comment]
> @Tantacrul @RomanPudashkin I have a potential fix for #11356 . I'll add that too

In a separate PR, please

**Pull request merged**